### PR TITLE
Setup Jupyter cluster for running with MPI

### PR DESCRIPTION
**Pull request by dalg24**
- Added mpi profile for ipyparallel.
- Made `MPI.COMM_SELF` the default parameter in the `EnergyStorageDevice` constructor

### CODE REVIEW

**Comment by dalg24** [comment]
Can one of the admins verify this patch?

**Comment by dalg24** [comment]
This comment is from Jenkins. Probably should make the server post under a bot account rather than myself...

**Comment by dalg24** [comment]
ok to test

**Comment by Rombur** [comment]
Is it ready to be merged or not?

File: docker/ipcluster_config.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import os
+c.IPClusterEngines.engine_launcher_class = 'MPI'
+if 'NPROC' in os.environ:
+    c.IPClusterEngines.n = int(os.environ['NPROC'])
```
Comment (Rombur) [inline docker/ipcluster_config.py]: Maybe use `CAP_NPROC` just in case there is a code that already uses `NPROCS`

**Comment by dalg24** [comment]
How about `JPY_ENGINES`?

**Comment by dalg24** [comment]
Or `JPY_CLUSTER_ENGINES`?

**Comment by Rombur** [comment]
Yeah that's fine. `NPROC` is just too general. If another code already uses it, it will be too hard to understand what's going on.

**Comment by dalg24** [comment]
Done

**Pull request merged**